Main.py

```diff
@@ -2,431 +2,368 @@
         conn.commit()
+        await ctx.send("```css\nScore added.\n```")
+
+        # calculate new avg
+        times_list = c.execute('SELECT Score, Date FROM Scores WHERE ID = ?', (member.id,)).fetchall()
+        reg_vals = []
+        sat_vals = []
+        for score, score_date in times_list:
+            (sat_vals if _get_day_from_ymd(score_date) == "Sat" else reg_vals).append(int(score))
+
+        avgs_list = c.execute('SELECT RegAvg, SatAvg FROM Ranking WHERE ID = ?', (member.id,)).fetchall()
+
+        def _format_avg_delta(ad):
+            return f"[{ad}]" if ad else ""
+
+        msg = "```"
+
+        if reg_vals:
+            new_reg_avg = int(statistics.mean(reg_vals))
+            old_reg_avg = avgs_list[0][0] if avgs_list else None
+            avg_diff = f"{str(new_reg_avg - old_reg_avg):+d}" if old_reg_avg is not None and day != "Sat" else None
+            msg += f"~ {member.name}'s Regular Crossword Avg: {_format_time(new_reg_avg)} " \
+                   f"{_format_avg_delta(avg_diff)}~\n"
+            c.execute(
+                "INSERT OR REPLACE INTO Ranking VALUES (:id, :name, :reg_avg, "
+                "(SELECT SatAvg FROM Ranking WHERE ID=:id))",
+                {"id": member.id, "name": member.name, "reg_avg": new_reg_avg}
+            )
+            conn.commit()
 
-    msg += "```"
-    await self.bot.say(msg)
-    conn.close()
+        if sat_vals:
+            new_sat_avg = int(statistics.mean(sat_vals))
+            old_sat_avg = avgs_list[0][1] if avgs_list else None
+            avg_diff = f"{str(new_sat_avg - old_sat_avg):+d}" if old_sat_avg is not None and day == "Sat" else None
+            msg += f"~ {member.name}'s Saturday Crossword Avg: {_format_time(new_sat_avg)} " \
+                   f"{_format_avg_delta(avg_diff)}~"
+            c.execute(
+                "INSERT OR REPLACE INTO Ranking VALUES(:id, :name, (SELECT RegAvg FROM Ranking WHERE ID=:id), "
+                ":new_sat_avg)",
+                {"id": member.id, "name": member.name, "sat_avg": new_sat_avg}
+            )
+            conn.commit()
+
+        msg += "```"
+        await ctx.send(msg)
+
+    finally:
+        conn.close()
 
-@bot.command(pass_context=True)
-@commands.has_role("crosswords")
-@asyncio.coroutine
-async def ltimes(self):
+
+@bot.command()
+@commands.has_role(CROSSWORD_ROLE)
+async def ltimes(ctx):
     """
     List your 20 most recent scores
     """
+
     conn = sqlite3.connect(DB_PATH)
-    c = conn.cursor()
-    timeslist = c.execute('SELECT Score,Date FROM Scores WHERE ID=?',
-                           (self.message.author.id,)).fetchall()
-    if not timeslist:
-        await self.bot.say('```No times found.```')
+
+    try:
+        c = conn.cursor()
+        times_list = c.execute("SELECT Score,Date FROM Scores WHERE ID=? ORDER BY Date DESC",
+                               (ctx.author.id,)).fetchall()
+        if not times_list:
+            await ctx.bot.say('```No times found.```')
```

Comment:
Should be `ctx.send`

Reply:
19e11bd1e01522040608edc461f68e8d7edf8cc6